### ISSUE TITLE
Write the GPX File to a New Folder each Month

### ISSUE THREAD

**Issue by Ib2412** [mendhak/gpslogger#1085]
ist es möglich die GPX Datei Automatisch in einen Ordner mit Jahr und Monat schreiben zu lassen.
Beispiel
2023/01
2023/02
ich lasse den Logger jeden Tag mitlaufen wenn ich das Haus verlasse, was dann ca. 330 Datein pro Jahr ergibt.
sehr lange zu scrollen und unübersichtlich.

Translate by Google

Is it possible to write the GPX file automatically to a folder with the year and month.
Example
2023/01
2023/02
I run the logger every day when I leave the house, which makes about 330 files per year.
Start with Macro Droid 
very long to scroll and confusing.

**Comment by mendhak** [comment]
I think this can be accomplished by creating directories if a `/` is entered into the custom file name. I have made a change in [this pull request](https://github.com/mendhak/gpslogger/pull/1086/files#diff-b0d81f55cb82de40cf3133469170a567c6f5684ec00522fedd1408ec4e39f7d2R58).

**Comment by Ib2412** [comment]
Exampel
/%year %Month %day
No dir ist createt

**Comment by mendhak** [comment]
Unless you've compiled the code yourself that won't work just yet, it's only in pull request stage. 
The idea will be that using something like `%Year/%Month/%day` should create a '2023/07' folder.

**Comment by Ib2412** [comment]
Thankyou I ist a Idee for the next Version.Thankyou for the great App --Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.Am 07.07.23, 15:48 schrieb mendhak ***@***.***>:

   
  Unless you've compiled the code yourself that won't work just yet, it's only in pull request stage. The idea will be that using something like %Year/%Month/%day should create a '2023/07' folder.

**Comment by mendhak** [comment]
If anyone wants to test, I have placed a prerelease APK here: https://github.com/mendhak/gpslogger/releases/tag/v128-rc1

This APK is not compatible with the F-Droid version, so if you install this you will lose all your previous GPSLogger settings/files etc.

**Comment by Ib2412** [comment]
einmal wurde ein Ordner erzeugt bei Eingabe von
/%year
dann habe ich versucht
/%year/%month
ab dann ging nichts mehr auch /%year ging nicht mehr.
und schwarzes design geht nicht

Translate by Google

once a folder was created when entering from
/%year
then I tried
/%year/%month
From then on nothing worked anymore, also /%year didn't work anymore.
and black design is not possible

**Comment by mendhak** [comment]
It's working fine for me, can you show how you've entered the filename. 

I first did a `/%year` which would create 2023.gpx. 

![image](https://github.com/mendhak/gpslogger/assets/746276/1a752bfa-34eb-430c-b808-22422cee575b)

Then did a `/%year/%month` which would create `2023/07.gpx`

![image](https://github.com/mendhak/gpslogger/assets/746276/5fab6bbd-2a71-46c4-bcc3-1840a27ac745)

**Comment by mendhak** [comment]
I found a problem, that the sharing dialogs show just the top level folder name, but not the files inside them. I'll have to find a way to list them. 

But now I'm seeing that the automated file uploads aren't working. All of the upload targets will need to be worked to get the subfolder+file and upload them. And I wonder if the subfolder needs to be created in the target upload area too. That's quite a lot of work :thinking:

**Comment by mendhak** [comment]
Unfortunately I've had to roll this back. I gave this a fair attempt over the past two weeks in the codebase, but this proved to be too complicated and headache inducing to implement. The current file and folder pattern is embedded in too many places and adding the extra logic needed to recognize another form of folders was causing too many problems especially with auto sending sharing and writing, with how dynamic it was. Files will need to be the only thing that can stay dynamic. 

Sorry for you this means the problem won't be solved, the best I can suggest is to try to form a file pattern that helps with organizing the files better, eg year_month_date so that it's at least easier to sort.
